src/annotator/util/test/buckets-test.js

```diff
@@ -1,44 +1,65 @@
-import { findClosestOffscreenAnchor, anchorBuckets } from '../buckets';
+import {
+  findClosestOffscreenAnchor,
+  computeAnchorPositions,
+  computeBuckets,
+} from '../buckets';
 import { $imports } from '../buckets';
 
-let tagIndex = 0;
-function fakeAnchorFactory(
-  offsetStart = 1,
-  offsetIncrement = 100,
-  boxHeight = 50
-) {
-  let highlightIndex = offsetStart;
-  return () => {
-    // In a normal `Anchor` object, `highlights` would be an array of
-    // DOM elements. Here, `highlights[0]` is the vertical offset (top) of the
-    // fake anchor's highlight box and `highlights[1]` is the height of the
-    // box. This is in used in conjunction with the mock for
-    // `getBoundingClientRect`, below
-    const anchor = {
-      annotation: { $tag: `t${tagIndex++}` },
-      highlights: [highlightIndex, boxHeight],
-    };
-    highlightIndex = highlightIndex + offsetIncrement;
-    return anchor;
-  };
-}
-
 describe('annotator/util/buckets', () => {
-  let fakeGetBoundingClientRect;
-
   let fakeAnchors;
+  let fakeAnchorPositions;
+  let fakeGetBoundingClientRect;
   let stubbedInnerHeight;
 
   beforeEach(() => {
-    const fakeAnchor = fakeAnchorFactory();
```

Comment:
In addition to the functional changes in this PR, it looks like there has been some general refactoring to the approach the tests use. Reviewing this code was more difficult because it wasn't clear to me initially how the changes related to the functional changes in the code. Would be very helpful here is to call out the non-functional changes in the PR description, a commit message or comments on the relevant parts of the PR.

Reply:
Yes, sorry. I should have suggest this change in a follow up.

The reasoning for the change was that the `fakeAnchor` utility created an structure that was simple to recreate. I feel looking at  `{ annotation: { $tag: 't0' }, highlights: [1, 50] }, ...` is simpler to know the `fakeAnchors` structure.